### ISSUE TITLE
New rule: consistent-type-import

### ISSUE THREAD

**Issue by rfermann** [typescript-eslint/typescript-eslint#1985]
**Repro**

```TS
import type { FooType } from './foo-type';
import { FooType } from './foo-type';

const fooType: import("foo-type").FooType = {}
```
All 3 lines above are valid ways of importing types.

**Expected Result**
A rule informing about the mix of different ways of type importing can be helpful to keep a consistent way of importing types over the project.

**Actual Result**
As for my knowledge there is no rule available for this purpose.

**Comment by David-Else** [comment]
Is there any reason to not always use the new `import type` syntax? It seems to be the most succinct.

**Comment by bradzacher** [comment]
It means you have to separate your type imports from your value imports:
```ts
import type { MyInterface } from 'foo';
import { MyValue } from 'foo';
```

Also for people that don't use babel, in 99.9% of imports, it won't provide any tangible benefits.

**Comment by David-Else** [comment]
I like the fact you have to separate your type imports from your value imports, makes it cleaner.

**Comment by bradzacher** [comment]
Merging into #2200
